### PR TITLE
Fix app.dock.setIcon() exception on platforms that do not have app.dock

### PR DESCRIPTION
**Pull request by jhurliman**

### CODE REVIEW

File: desktop/index.ts [hunk 1]
```diff
@@ -298,8 +298,10 @@ app.on("ready", async () => {
 
     // In development, we run with the pre-packaged Electron binary, so we need to manually set the Dock icon.
     try {
-      // This fails when opening the app from a packaged DMG.
-      app.dock.setIcon("resources/icon/icon.png");
+      if (app.dock != undefined) {
```
Comment (defunctzombie) [inline desktop/index.ts:301]: Bad typescript typing info if app.dock is sometimes undefined?

Comment (jtbandes) [reply desktop/index.ts:301]: `app.dock?.setIcon(...)`

Comment (jtbandes) [reply desktop/index.ts:301]: Also wait - what is going on here? We are already inside a `try` ....

Comment (jhurliman) [reply desktop/index.ts:301]: or-undefined

:sigh:

**Pull request merged**